conformance_tests/test_tiovx/test_safe_casts.c

```diff
@@ -0,0 +1,624 @@
+            break;
+        }
+        case VX_TYPE_IMAGE:
+        {
+            vx_image image = vxCastRefAsImage(ref, &status);
+            break;
+        }
+        case VX_TYPE_KERNEL:
+        {
+            vx_kernel kernel = vxCastRefAsKernel(ref, &status);
+            break;
+        }
+        case VX_TYPE_LUT:
+        {
+            vx_lut lut = vxCastRefAsLUT(ref, &status);
+            break;
+        }
+        case VX_TYPE_MATRIX:
+        {
+            vx_matrix matrix = vxCastRefAsMatrix(ref, &status);
+            break;
+        }
+        case VX_TYPE_NODE:
+        {
+            vx_node node = vxCastRefAsNode(ref, &status);
+            break;
+        }
+        case VX_TYPE_OBJECT_ARRAY:
+        {
+            vx_object_array object_array = vxCastRefAsObjectArray(ref, &status);
+            break;
+        }
+        case VX_TYPE_PARAMETER:
+        {
+            vx_parameter parameter = vxCastRefAsParameter(ref, &status);
+            break;
+        }
+        case VX_TYPE_PYRAMID:
+        {
+            vx_pyramid pyramid = vxCastRefAsPyramid(ref, &status);
+            break;
+        }
+        case VX_TYPE_REMAP:
+        {
+            vx_remap remap = vxCastRefAsRemap(ref, &status);
+            break;
+        }
+        case VX_TYPE_SCALAR:
+        {
+            vx_scalar scalar = vxCastRefAsScalar(ref, &status);
+            break;
+        }
+        case VX_TYPE_TENSOR:
+        {
+            vx_tensor tensor = vxCastRefAsTensor(ref, &status);
+            break;
+        }
+        case VX_TYPE_THRESHOLD:
+        {
+            vx_threshold threshold = vxCastRefAsThreshold(ref, &status);
+            break;
+        }
+        case VX_TYPE_USER_DATA_OBJECT:
+        {
+            vx_user_data_object user_data_object = vxCastRefAsUserDataObject(ref, &status);
+            break;
+        }
+        default:
+            status = VX_ERROR_INVALID_TYPE;
+    }
+    if (ref) {EXPECT_EQ_VX_STATUS(VX_SUCCESS, vxReleaseReference(&ref));}
+    return status;
+}
+
+/* Check to see if the type is currently supported */
+vx_bool is_supported(vx_enum type)
+{
+    switch (type)
+    {
+        case VX_TYPE_ARRAY:
+        case VX_TYPE_CONTEXT:
+        case VX_TYPE_CONVOLUTION:
+        case VX_TYPE_DELAY:
+        case VX_TYPE_DISTRIBUTION:
+        case VX_TYPE_GRAPH:
+        case VX_TYPE_IMAGE:
+        case VX_TYPE_KERNEL:
+        case VX_TYPE_LUT:
+        case VX_TYPE_MATRIX:
+        case VX_TYPE_NODE:
+        case VX_TYPE_OBJECT_ARRAY:
+        case VX_TYPE_PARAMETER:
+        case VX_TYPE_PYRAMID:
+        case VX_TYPE_REMAP:
+        case VX_TYPE_SCALAR:
+        case VX_TYPE_TENSOR:
+        case VX_TYPE_THRESHOLD:
+        case VX_TYPE_USER_DATA_OBJECT:
+            return vx_true_e;
+        default:
+            return vx_false_e;
+   }
+}
+
+TEST(tivxSafeCasts, testCastGetFromSomeType)
+{
+    int i;
+    vx_graph graph = vxCreateGraph(context_->vx_context_);
+    ASSERT_VX_OBJECT(graph);
```

Comment:
It looks like the ASSERT_VX_OBJECT requires two arguments, the second being the OpenVX type